contracts/lib/PreciseUnitMath.sol

```diff
@@ -13,15 +13,15 @@
     See the License for the specific language governing permissions and
     limitations under the License.
 
-    SPDX-License-Identifier: Apache License, Version 2.0
+    SPDX-License-Identifier: Apache-2.0
 */
 
-pragma solidity 0.6.10;
-pragma experimental ABIEncoderV2;
+pragma solidity 0.8.19;
 
-import { SafeCast } from "@openzeppelin/contracts/utils/SafeCast.sol";
-import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
-import { SignedSafeMath } from "@openzeppelin/contracts/math/SignedSafeMath.sol";
+
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
```

Comment:
Do we still need `SafeMath` with compiler 0.8 ? Could create a huge, unnecessary diff though.

Reply:
We could remove SafeMath as the overflow/underflow checks are now inbuilt. The SafeMath lib now provides nice abstractions like tryAdd etc. That can be a future TODO to remove SafeMath where possible